api/src/test/java/org/openmrs/api/OrderServiceTest.java

```diff
@@ -3109,5 +3109,52 @@ public void saveOrder_shouldRevisePreviousNonCodedOrderIfItIsAlreadyExisting() t
 		Assert.assertTrue(saveOrder.getAsNeeded());
 		assertNotNull(orderService.getOrder(saveOrder.getOrderId()));
 	}
-	
+
+	@Test
+	public void saveRetrospectiveOrder_shouldDiscontinueOrderInRetrospectiveEntry() throws Exception {
+		executeDataSet("org/openmrs/api/include/OrderServiceTest-ordersWithAutoExpireDate.xml");
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss.S");
+		Date originalOrderDateActivated = dateFormat.parse("2008-11-19 09:24:10.0");
+		Date discontinuationOrderDate = DateUtils.addDays(originalOrderDateActivated, 2);
+
+		Order originalOrder = orderService.getOrder(201);
+		assertNull(originalOrder.getDateStopped());
+		assertEquals(dateFormat.parse("2008-11-23 09:24:09.0"), originalOrder.getAutoExpireDate());
+		assertFalse(originalOrder.isActive());
+		assertTrue(originalOrder.isActive(discontinuationOrderDate));
+
+		Order discontinueationOrder = originalOrder.cloneForDiscontinuing();
+		discontinueationOrder.setPreviousOrder(originalOrder);
+		discontinueationOrder.setEncounter(encounterService.getEncounter(17));
+		discontinueationOrder.setOrderer(providerService.getProvider(1));
+		discontinueationOrder.setDateActivated(discontinuationOrderDate);
+		orderService.saveRetrospectiveOrder(discontinueationOrder, null);
+
+		assertNotNull(originalOrder.getDateStopped());
+		assertEquals(discontinueationOrder.getAutoExpireDate(), discontinueationOrder.getDateActivated());
+	}
+
+	@Test
+	public void saveRetrospectiveOrder_shouldDiscontinueAndStopActiveOrderInRetrospectiveEntry() throws Exception {
```

Comment:
I see no difference between this test and the one above except the test method name